font-patcher

```diff
@@ -36,6 +36,614 @@ except ImportError:
+            b |= ITALIC
+        # Regular is just the basic weight
+        if len(self.weight_token) == 0:
+            b |= REGULAR
+        b |= WWS # We assert this by our naming process
+        return b
+
+    def rename_font(self, font):
+        """Rename the font to include all information we found (font is fontforge font object)"""
+        font.fontname = self.ps_fontname()
+        font.fullname = self.fullname()
+        font.familyname = self.ps_familyname()
+
+        # We have to work around several issues in fontforge:
+        #
+        # a. Remove some entries from SFNT table; fontforge has no API function for that
+        #
+        # b. Fontforge does not allow to set SubFamily (and other) to any value:
+        #
+        #    Fontforge lets you set any value, unless it is the default value. If it
+        #    is the default value it does not set anything. It also does not remove
+        #    a previously existing non-default value. Why it is done this way is
+        #    unclear:
+        #      fontforge/python.c SetSFNTName() line 11431
+        #        return( 1 ); /* If they set it to the default, there's nothing to do */
+        #
+        #    Then is the question: What is the default? It is taken from the
+        #    currently set fontname (??!). The fontname is parsed and everything
+        #    behind the dash is the default SubFamily:
+        #      fontforge/tottf.c DefaultTTFEnglishNames()
+        #      fontforge/splinefont.c _GetModifiers()
+        #
+        #    To fix this without touching Fontforge we need to set the SubFamily
+        #    directly in the SFNT table:
+        #
+        # c. Fontforge has the bug that it allows to write empty-string to a SFNT field
+        #    and it is actually embedded as empty string, but empty strings are not
+        #    shown if you query the sfnt_names *rolleyes*
+
+        sfnt_list = []
+        TO_DEL = ['Family', 'SubFamily', 'Fullname', 'Postscriptname', 'Preferred Family',
+                'Preferred Styles', 'Compatible Full', 'WWS Family', 'WWS Subfamily']
+        for l, k, v in list(font.sfnt_names):
+            if not k in TO_DEL:
+                sfnt_list += [( l, k, v )]
+
+        sfnt_list += [( 'English (US)', 'Family', self.family() )]
+        sfnt_list += [( 'English (US)', 'SubFamily', self.subfamily() )]
+        sfnt_list += [( 'English (US)', 'Fullname', self.fullname() )]
+        sfnt_list += [( 'English (US)', 'PostScriptName', self.psname() )]
+
+        p_fam = self.preferred_family()
+        if len(p_fam):
+            sfnt_list += [( 'English (US)', 'Preferred Family', p_fam )]
+        p_sty = self.preferred_styles()
+        if len(p_sty):
+            sfnt_list += [( 'English (US)', 'Preferred Styles', p_sty )]
+
+        font.sfnt_names = tuple(sfnt_list)
+
+        font.macstyle = self.macstyle(0)
+        font.os2_stylemap = self.fs_selection(0)
+
+############################################
```

Comment:
Ha! This is funny, because we had this before:

![image](https://user-images.githubusercontent.com/16012374/152677567-bb9294df-550a-4489-ab09-d193a25caec1.png)

We can easily re-introduce this be reverting commit df42c917128.

Here is the commit message
```
[why]
A lot people expect the font-patcher to be a stand alone script. They
even think that the source glyphs (symbols) to be added to be somehow
magically there and one PR makes sure that they are fetched if missing.

The same problem arises when we have a script distributed over multiple
files. For maintenance reasons and code quality this is what one wants.
But that might hinder easy use of the font-patcher.
```

But probably you are right, and more files are better (as I thought in the beginning).
`font-patcher` could still fetch the missing files (like implemented with the `--download-source-glyphs` or how I called it), to simplify for 'download the one script' users.

I was at that time not aware of your `font-patcher.zip` project, which would also solve that in a nice way.

Yes, as I write this, we should probably revert df42c917128 and add some guard code around it.
The big code blob in one file is really not nice and nothing I would do at work.

Will push soon.